dashboard/src/components/DeploymentStatus/DeploymentStatus.tsx

```diff
@@ -1,25 +1,33 @@
 import * as React from "react";
 
 import { AlertTriangle } from "react-feather";
+import isSomeResourceLoading from "../../components/AppView/helpers";
 import Check from "../../icons/Check";
 import Compass from "../../icons/Compass";
 import { hapi } from "../../shared/hapi/release";
-import { IDeploymentStatus, IResource } from "../../shared/types";
+import { IDeploymentStatus, IKubeItem, IResource } from "../../shared/types";
 import "./DeploymentStatus.css";
 
 interface IDeploymentStatusProps {
-  deployments: IResource[];
+  deployments: Array<IKubeItem<IResource>>;
   info?: hapi.release.IInfo;
 }
 
 class DeploymentStatus extends React.Component<IDeploymentStatusProps> {
   public render() {
+    if (isSomeResourceLoading(this.props.deployments)) {
+      return this.renderLoading();
+    }
     if (this.props.info && this.props.info.deleted) {
       return this.renderDeletedStatus();
     }
     return this.isReady() ? this.renderSuccessStatus() : this.renderPendingStatus();
   }
 
+  private renderLoading() {
+    return <span className="DeploymentStatus">Loading...</span>;
```

Comment:
having a function for this only seems an overkill

Reply:
I just followed the style of this file. This specific method may be an overkill but not doing it may be inconsistent. Anyway, I am fine changing this.